torchtune/datasets/_wikitext.py

```diff
@@ -4,46 +4,59 @@
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
-from typing import Any, Dict, Optional
+from typing import Any, Dict, Optional, Union
 
-from torchtune.datasets._text_completion import TextCompletionDataset
+from torchtune.datasets._packed import PackedDataset
+
+from torchtune.datasets._text_completion import (
+    text_completion_dataset,
+    TextCompletionDataset,
+)
 
 from torchtune.modules.tokenizers import ModelTokenizer
 
 
 def wikitext_dataset(
     tokenizer: ModelTokenizer,
-    source: str = "wikitext",
-    subset: str = "wikitext-103-raw-v1",
+    source: str = "EleutherAI/wikitext_document_level",
+    subset: str = "wikitext-103-v1",
     max_seq_len: Optional[int] = None,
+    packed: bool = False,
     split: str = "train",
     **load_dataset_kwargs: Dict[str, Any],
-) -> TextCompletionDataset:
+) -> Union[TextCompletionDataset, PackedDataset]:
     """
-    Support for family of datasets similar to `wikitext <https://huggingface.co/datasets/wikitext>`_,
-    an unstructured text corpus consisting of articles from Wikipedia.
+    Support for family of datasets similar to `wikitext
+    <https://huggingface.co/datasets/EleutherAI/wikitext_document_level>`_,
+    an unstructured text corpus consisting of fulls articles from Wikipedia.
 
     Args:
         tokenizer (ModelTokenizer): Tokenizer used by the model that implements the ``tokenize_messages`` method.
-        source (str): path string of dataset, anything supported by Hugging Face's ``load_dataset``
+        source (str): path to dataset repository on Hugging Face. For local datasets,
+            define source as the data file type (e.g. "json", "csv", "text") and pass
+            in the filepath in ``data_files``. See Hugging Face's ``load_dataset``
             (https://huggingface.co/docs/datasets/en/package_reference/loading_methods#datasets.load_dataset.path)
-        subset (str): name of subset of data to use, see the `wikitext page <https://huggingface.co/datasets/wikitext#data-fields>`_
-            for available subsets.
+            for more details.
+        subset (str): name of subset of data to use, see the `wikitext page
+            <https://huggingface.co/datasets/EleutherAI/wikitext_document_level#data-instances>`_
+            for available subsets. Default is ``"wikitext-103-v1"``.
         max_seq_len (Optional[int]): Maximum number of tokens in the returned input and label token id lists.
             Default is None, disabling truncation. We recommend setting this to the highest you can fit in memory
             and is supported by the model. For example, llama2-7B supports up to 4096 for sequence length.
+        packed (bool): Whether or not to pack the dataset to ``max_seq_len`` prior to training. Default is False.
```

Comment:
@RdoubleA spotted that this isn't used in wikitext, is it just to match the signature of text completion builder?

Reply:
oh this should be passed into `text_completion_dataset`. will update.